src/Planner/Planner.cpp

```diff
@@ -1341,27 +1341,52 @@ void Planner::buildPlanForQueryNode()
 
     const auto & settings = query_context->getSettingsRef();
 
-    /// Check support for JOIN for parallel replicas with custom key
-    if (planner_context->getTableExpressionNodeToData().size() > 1)
+    if (settings.allow_experimental_parallel_reading_from_replicas > 0)
```

Comment:
I'd rather use [Context::canUseParallelReplicas()]( https://github.com/ClickHouse/ClickHouse/blob/500a281acd6eb764dd6469e3249c87fd1139ebf7/src/Interpreters/Context.cpp#L5014C15-L5014C37)

Reply:
Makes sense, although that function name is bad. It says `ParallelReplicas` when it should be explicitly referring to task-based parallel replicas (which are the ones that we want to disable here)